### PR TITLE
GitHub: build release binaries and upload to release

### PR DESCRIPTION
**Pull request by guggero**
This commit adds another GitHub workflow that is activated for each
pushed tag. The release binaries are compiled from that tag for all
supported architectures. A new release in the GitHub repository is then
created for the tag and the finished binary packages are uploaded to
that release.

Example build can be seen here: https://github.com/guggero/lnd/runs/1423819682?check_suite_focus=true
Result can be seen here: https://github.com/guggero/lnd/releases/tag/v0.0.0-beta.test3 (taken out of draft mode manually for it to be visible to anyone).

Before the manual edit, it looked like this (confirming draft mode):
gh-actions-release-draft

### CODE REVIEW

**Comment by Roasbeef** [comment]
Just to confirm: the release won't be auto-published and instead will remain in the "draft" state, correct?

**Comment by Roasbeef** [comment]
I ask as we'll still want them just to sit in the draft state so we can add release notes, signatures, etc.

**Comment by guggero** [comment]
Very good question, @Roasbeef. It indeed wasn't put into draft mode which is not what we want at all.

I used another action now and was able to even include our default text body with the versions (release and golang) automatically being replaced. This should shave off another 5 minutes of manual, error prone work for each release.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ env.RELEASE_VERSION }}
+          name: lnd ${{ env.RELEASE_VERSION }}
+          draft: true
+          prerelease: false
+          files: lnd-${{ env.RELEASE_VERSION }}/*
+          body: |
```
Comment (Roasbeef) [inline .github/workflows/release.yaml:57]: This is so dope! As is we'll need to sync this up with the template periodically, but a small amount of manual work required for something that changes so rarely, so I think it's OK.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        run: make release tag=${{ env.RELEASE_VERSION }}
+
+      - name: Create Release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ env.RELEASE_VERSION }}
+          name: lnd ${{ env.RELEASE_VERSION }}
+          draft: true
```
Comment (Roasbeef) [inline .github/workflows/release.yaml:54]: 👍

**Review by Roasbeef — APPROVED** [review]
Excellent work on this! Now all we need to do is have developers verify locally they arrive at the same build, then upload their sigs!

**Pull request merged**